packages/contracts/contracts/AnonAadhaar.sol

```diff
@@ -0,0 +1,58 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.19;
+
+import "../interfaces/IAnonAadhaarVerifier.sol";
+import "../interfaces/IAnonAadhaar.sol";
+
+contract AnonAadhaar is IAnonAadhaar {
+    address public verifier;
+    uint256 public pubkeyHash;
+
+    constructor(address _verifier, uint256 _pubkeyHash) {
+        verifier = _verifier;
+        pubkeyHash = _pubkeyHash;
+    }
+
+    /// @dev Verifies that the public key received is corresponding with the one stored in the contract.
+    /// @param _receivedpubkeyHash: Public key received.
+    /// @return Verified bool
+    function verifyPublicKey(uint256 _receivedpubkeyHash) private view returns (bool) {
+        return pubkeyHash == _receivedpubkeyHash;
+    }
+
+    /// @dev Verifies that the signal received is corresponding with the one used in proof generation.
+    /// @param _receivedSignal: Signal received.
+    /// @param signalHash: Hash of the signal from Proof.
+    /// @return Verified bool
+    function verifySignalHash(uint256 _receivedSignal, uint256 signalHash) public pure returns (bool) {
```

Comment:
I agree that the signal is just a commitment and it can be used any hash function. But as we want to ensure that the resulting hash is constrained by the snark field, I thought it's better to provide the hashing method as part of the SDK/Protocol, making the life of devs easier. 

Also while you can just send the signal hashed to the verifier you might want to check the data that has been committed while generating the proof, to prevent it from front-running, that's why I added the hash method so any contract above can simply send to the verifier the plain signal, after having it checked it, and the verifier contract will then compute the hash of the committed value to verify the gorth16 proof.
This way the check of the plain signal value is delegated to the contract logic and not to the verifier, making it easier for devs to use `signal` as they want for their use case.

Reply:
Ok, got it 👍